File: RUBY/zadanie_3.rb
input = gets.chomp.split
output = []
input.each_with_index do |number, index|
  sum = number.chars.map(&:to_i).sum
  product = number.chars.map(&:to_i).reduce(:*)
  output << index if sum < product
end
puts output.join(" ")

File: RUST/zadanie_3.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let numbers: Vec<&str> = input.trim().split_whitespace().collect();
    let mut output = Vec::new();
    for (i, &number) in numbers.iter().enumerate() {
        let digits: Vec<i32> = number.chars().map(|d| d.to_digit(10).unwrap() as i32).collect();
        let sum: i32 = digits.iter().sum();
        let product: i32 = digits.iter().product();
        if sum < product {
            output.push(i);
        }
    }
    for index in output {
        print!("{} ", index);
    }
}
